### PR TITLE
Database tree-shake step: modularize CompoundWrite

### PR DESCRIPTION
**Pull request by hsubox76**
Extract methods from `CompoundWrite`.

### CODE REVIEW

File: packages/database/src/core/CompoundWrite.ts [hunk 1]
```diff
@@ -30,145 +30,170 @@ import { each } from './util/util';
...
 
   static empty(): CompoundWrite {
     return new CompoundWrite(new ImmutableTree(null));
   }
+}
 
-  addWrite(path: Path, node: Node): CompoundWrite {
-    if (path.isEmpty()) {
-      return new CompoundWrite(new ImmutableTree(node));
+export function addWrite(
```
Comment (schmidt-sebastian) [inline packages/database/src/core/CompoundWrite.ts:40]: Can you prefix all these method names with `compoundWrite`? This has some benefits:
- All imports sort in order
- It is still possible to translate Android and iOS APIs to Web APIs
- There are no clashes and we don't have to think about how to deal with clashes on a case-by-case basis.
- 
I initially did not like this method naming either, but we updated all of Firestore to match this pattern. This created a very consistent method-naming convention across the SDK. See https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/src/local/local_store_impl.ts#L216

File: packages/database/src/core/WriteTree.ts [hunk 2]
```diff
@@ -215,34 +228,34 @@ export class WriteTree {
...
     if (!writeIdsToExclude && !includeHiddenWrites) {
-      const shadowingNode = this.visibleWrites_.getCompleteNode(treePath);
+      const shadowingNode = getCompleteNode(this.visibleWrites_, treePath);
       if (shadowingNode != null) {
         return shadowingNode;
       } else {
-        const subMerge = this.visibleWrites_.childCompoundWrite(treePath);
-        if (subMerge.isEmpty()) {
+        const subMerge = childCompoundWrite(this.visibleWrites_, treePath);
+        if (isEmpty(subMerge)) {
```
Comment (schmidt-sebastian) [inline packages/database/src/core/WriteTree.ts:236]: Especially this name is a bit overly generic.

File: packages/database/src/core/CompoundWrite.ts (hunk 1 above)
Comment (hsubox76) [reply packages/database/src/core/CompoundWrite.ts]: Done. It's starting to look a little like Java now!

**Review by schmidt-sebastian — APPROVED** [review]
Thanks!

**Pull request merged**